### DISCUSSION TITLE
Adding a test action

### DISCUSSION THREAD

**Discussion by learn-more** [premake/premake-core#2189] [Q&A]
Currently I have added a test action like this, for use in github runners,
where 'test-project' is a c++ application with gtest testcases:

```lua
newoption {
    trigger     = "configuration",
    value       = "Release",
    description = "The configuration to use",
    default     = "Release",
    category    = "Build/Test Options",
    allowed = {
       { "Release",     "The optimized version" },
       { "Debug",       "The debug version" },
    }
}

newaction {
    trigger = "test",
    description = "Test the project using the specified platform and configuration",
    execute = function()
        for _, platform in pairs(selected_platforms()) do
            printf('Testing %s|%s', platform, _OPTIONS.configuration)
            local proj = project('test-project')
            local config = proj.configs[_OPTIONS.configuration .. platform]
            local target = config.buildtarget
            local res = os.execute(target.fullpath)
            if not res then 
                os.exit(1)
            end
        end
    end,
}
```

Is this a decent / robust way of running the tests?
I am mainly interested in the selection of the config that I cobbled together here, is that acceptable, or is it prone to breaking?
Are there better / easier to use alternatives to this?

**Comment by Jarod42** [comment]
I tend to use several generators, so your action won't fit my usage ;-)

> for use in github runners

So I would just do in github action something like:
```
premake5 gmake2
make config=release
bin/Release/test-project
make config=debug
bin/Debug/test-project
```
or use github matrix.

But if your script fits your needs, keep it :-)

> or is it prone to breaking?

`proj.configs[_OPTIONS.configuration .. platform]` seems fragile if premake's internal mapping changes.
